profile/profile.go

```diff
@@ -26,11 +26,12 @@ type profileSource string
 
 // Profile Sources.
 const (
-	SourceLocal      profileSource = "local"   // local, editable
-	SourceSpecial    profileSource = "special" // specials (read-only)
-	SourceNetwork    profileSource = "network"
-	SourceCommunity  profileSource = "community"
-	SourceEnterprise profileSource = "enterprise"
+	SourceLocal              profileSource = "local"   // local, editable
+	SourceSpecial            profileSource = "special" // specials (read-only)
+	SourceNetwork            profileSource = "network"
+	SourceCommunity          profileSource = "community"
+	SourceEnterprise         profileSource = "enterprise"
+	SourceEnvironmentDefined profileSource = "environment-defined" // defined from environment variable
```

Comment:
Ok, so GitHub ditched my longest comment here. Ugh. Here we go again...

I thought about the whole use case of this feature and how it interacts with other features.
If we strictly scope profiles from this feature to a separate source, it will not be able to interact with other features.

For example, @ppacher suggested a feature, where one could provision profiles/settings by placing them in a configured directory on disk. Portmaster would then search the profiles and apply them if one matched. These could live in a scope name `provision-dir`.
These profiles however, would be inaccessible to use via the environment variable.

My suggestion is:
- Create a new generic source called `user`, which holds profiles created by the user. (Also, from the UI in the future.)
- If a profile ID is specified via the environment variable and no `/` is in the ID, we assume the source `user`.
- If there is a source defined, we load the profile from there, BUT we would not automatically create one there, but fail instead.

I think this is a nice small change that will help future-proof this feature.
What are your thoughs @vlabo and @ppacher?

Reply:
I really like the idea but IMHO we could leave the "environment-defined" source if we just add the support to specify the profile source in the env-variable. I.e. `my-custom-profile` would create a profile with the ID `env-var/my-custom-profile` while `file/my-custom-profile` would try to load the profile from the provisioning directory and fail if it does not exist. Likewise, using `local/0000-0000-0000-0000` would try to load the profile generated by the PM or fail otherwise. This way we don't need to introduce a somewhat one-for-all `user` source

Reply:
Also we should probably add a developer/advanced setting that enables the use of environment variables for profile selection?

Reply:
Let's go with:
- `env-var` source for profiles created from the environment variable (if not source is defined)
- Create an advanced setting that enabled environment variable profile selection
- profiles from the env var may only be created in the `env-var` scope, if loading from another source fails, it fails and all connection should be blocked (as the profile is missing).